atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1900,6 +1900,35 @@ private void putMetadataWithoutChangingSettings(final TableReference tableRef, f
         delete(AtlasDbConstants.DEFAULT_METADATA_TABLE, oldVersions);
     }
 
+
+    @Override
+    public void deleteRange(final TableReference tableRef, final RangeRequest range) {
+        if (range.equals(RangeRequest.all())) {
+            try {
+                clientPool.run(
```

Comment:
Yeah, I saw that. That's kind of lame in that it's M*N retries, but the retry for the top level one also retries all of the tables in a truncateTables(multiple tables) call, so it does have a different semantic.

I think I'm going to fix up truncateTables() proper version to just use .run(), I think it would make more sense.